pkgs/applications/office/foliate/default.nix

```diff
@@ -1,43 +1,70 @@
-{ stdenv, lib, fetchFromGitHub, meson, gettext, glib, gjs, ninja, python3, gtk3
-, webkitgtk_4_1, gsettings-desktop-schemas, wrapGAppsHook, desktop-file-utils
-, gobject-introspection, glib-networking }:
+{ stdenv
+, lib
+, fetchFromGitHub
+, meson
+, gettext
+, glib
+, gjs
+, ninja
+, python3
+, gtk4
+, webkitgtk_6_0
+, gsettings-desktop-schemas
+, wrapGAppsHook
+, desktop-file-utils
+, gobject-introspection
+, glib-networking
+, pkg-config
+, libadwaita
+, cmake
+}:
 
 stdenv.mkDerivation rec {
   pname = "foliate";
-  version = "2.6.4";
+  version = "3.0.1";
 
   src = fetchFromGitHub {
     owner = "johnfactotum";
     repo = pname;
     rev = version;
-    sha256 = "sha256-Pr2YA2MHXD4W7lyCxGAVLKyoZarZ8t92RSkWle3LNuc=";
+    hash = "sha256-Nw6ABV0kqVeF6PGWSSoIXV8O0ymAfhUmdmF++BWSCtM=";
   };
 
-  nativeBuildInputs = [ meson ninja python3 wrapGAppsHook gobject-introspection ];
-
   postPatch = ''
     patchShebangs build-aux/meson/postinstall.py
 
     substituteInPlace src/main.js \
       --replace "'WebKit2': '4.0'" "'WebKit2': '4.1'"
   '';
 
-  postFixup = ''
-    sed -i "1 a imports.package._findEffectiveEntryPointName = () => 'com.github.johnfactotum.Foliate';" $out/bin/.com.github.johnfactotum.Foliate-wrapped
-    ln -s $out/bin/com.github.johnfactotum.Foliate $out/bin/foliate
-  '';
+  nativeBuildInputs = [
+    gobject-introspection
+    meson
+    ninja
+    pkg-config
+    python3
+    wrapGAppsHook
+
+    cmake
+    libadwaita
```

Comment:
Runtime dependencies should go to `buildInputs`, see https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md.